Core/src/org/sleuthkit/autopsy/coreutils/SQLiteTableReader.java

```diff
@@ -0,0 +1,560 @@
+     * @param id   The input files id value
+     *
+     * @return The path of the file on disk
+     *
+     * @throws IOException            Exception writing file contents
+     * @throws NoCurrentCaseException Current case closed during file copying
+     */
+    private String copyFileToTempDirectory(AbstractFile file, long fileId)
+            throws IOException, NoCurrentCaseException {
+
+        String localDiskPath = Case.getCurrentCaseThrows().getTempDirectory()
+                + File.separator + fileId + file.getName();
+        File localDatabaseFile = new File(localDiskPath);
+        if (!localDatabaseFile.exists()) {
+            ContentUtils.writeToFile(file, localDatabaseFile);
+        }
+        return localDiskPath;
+    }
+
+    /**
+     * Executes the query and assigns resource references to instance variables.
+     *
+     * @param query Input query to execute
+     *
+     * @throws SQLiteTableReaderException
+     */
+    private void openTableResources(String query) throws SQLiteTableReaderException {
+        try {
+            ensureOpen();
+            statement = conn.prepareStatement(query);
+            queryResults = statement.executeQuery();
+            currentMetadata = queryResults.getMetaData();
+            totalColumnCount = currentMetadata.getColumnCount();
+            liveResultSet = true;
+        } catch (SQLException ex) {
+            throw new SQLiteTableReaderException(ex);
+        }
+    }
+
+    /**
+     * Ensures both the statement and the result set for a table are closed.
+     */
+    private void closeTableResources() {
+        try {
+            if (Objects.nonNull(statement)) {
+                statement.close();
+            }
+            if (Objects.nonNull(queryResults)) {
+                queryResults.close();
+            }
+            liveResultSet = false;
+        } catch (SQLException ex) {
+            //Do nothing, can't close.. tried our best.
+        }
+    }
+
+    /**
+     * Closes all resources attached to the database file.
+     *
+     * @throws SQLiteTableReaderException
+     */
+    @Override
+    public void close() throws SQLiteTableReaderException {
+        try {
+            closeTableResources();
+            if (Objects.nonNull(conn)) {
+                conn.close();
+            }
+        } catch (SQLException ex) {
+            throw new SQLiteTableReaderException(ex);
+        }
+    }
+
+    /**
+     * Provides status of the current read operation.
+     *
+     * @return
+     */
+    public boolean isFinished() {
+        return !liveResultSet;
+    }
+
+    /**
+     * Last ditch effort to close the connections during garbage collection.
+     *
+     * @throws Throwable
+     */
+    @Override
+    protected void finalize() throws Throwable {
+        try {
+            close();
+        } catch (SQLiteTableReaderException ex) {
+            //Do nothing, we tried out best to close the connection.
```

Comment:
It is fine to not propagate the exception here, but I would prefer that we log it.